met-web/src/components/engagement/form/EngagementWidgets/Documents/DocumentForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { Grid, Divider } from '@mui/material';
+import { MetHeader3, PrimaryButton, SecondaryButton, GreyButton } from 'components/common';
+import { Document } from 'models/document';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import BorderColorIcon from '@mui/icons-material/BorderColor';
+import { WidgetDrawerContext } from '../WidgetDrawerContext';
+import { WidgetType } from 'models/widget';
+import { DocumentsContext } from './DocumentsContext';
+
+const DocumentForm = () => {
+    const { handleWidgetDrawerOpen, widgets } = useContext(WidgetDrawerContext);
+    const { handleAddDocumentDrawerOpen, documents } = useContext(DocumentsContext);
+
+    const dispatch = useAppDispatch();
+    const [addedDocuments, setAddedDocuments] = useState<Document[]>([]);
+    const [savingWidgetItems, setSavingWidgetItems] = useState(false);
+
+    const widget = widgets.filter((widget) => widget.widget_type_id === WidgetType.Document)[0] || null;
+    useEffect(() => {
+        setAddedDocuments([]);
+    }, [documents, widget]);
+
+    const addDocument = async () => {
```

Comment:
let's clear these methods, maybe keep saveWidgetItems. This form will behave differently than the who is listening form. So let's clear out the states as the next task in line should handle them.

Reply:
ok will do that